src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -411,6 +411,33 @@ private BoundExpression BindToInferredDelegateType(BoundExpression expr, Binding
                 }
                 delegateType = CreateErrorType();
             }
+            else
+            {
+                // If synthesizing a delegate with `decimal`/`DateTime` default value,
+                // check that the corresponding `*ConstantAttribute` is available.
+                var delegateParameters = delegateType.DelegateParameters();
```

Comment:
>var delegateParameters = delegateType.DelegateParameters(); [](http://example.com/codeflow?start=16&length=59)

It looks like this code path is specific to local declarations. Are there other scenarios when synthesized delegates can be used?

Reply:
You're right, this is only for variable declarations. But it was such a nice place!